src/store/initial-state.js

```diff
@@ -62,40 +72,32 @@ export const preparePipelineState = props => {
  * will persist if the pipeline data is reset.
  * Merge local storage and add custom state overrides from props etc
  * @param {object} props Props passed to App component
+ * @return {object} Updated initial state
  */
 export const prepareNonPipelineState = props => {
   const state = mergeLocalStorage(createInitialState());
-  // Override flag defaults with URL values (on page load only)
-  state.flags = Object.assign({}, state.flags, getFlagsFromUrl());
-  // Override theme if set in props
-  if (props.theme) {
-    state.theme = props.theme;
-  }
-  // Override button visibility if set in props
-  if (props.visible) {
-    state.visible = Object.assign({}, state.visible, props.visible);
-  }
-  return state;
+
+  // Override with URL values
+  const flags = Object.assign({}, state.flags, getFlagsFromUrl());
```

Comment:
In all of these `Object.assign` cases, spread could be used to make this clearer e.g.

```
const flags = { ...state.flags, ...getFlagsFromUrl() };
```

Reply:
Personally I'm probably more pro-`Object.assign` than pro-spread, but yeah sure